### PR TITLE
Review

### PR DESCRIPTION
**Pull request by tavisca-cpadole**

### CODE REVIEW

File: AddToCart/Program.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        {
+            double totalPrice = 0;
+            foreach (var item in _cartItem.ItemList())
+            {
+                totalPrice += _cartItem.GetPrice(item.Key);
+            }
+            return totalPrice;
+        }
+
+        public double GetFinal()
```
Comment (tavisca-mswarnesh) [inline AddToCart/Program.cs]: please use appropriate name

File: AddToCart/Program.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            {
+                _cartItems[product] += 1;
+            }
+            else
+            {
+                _cartItems[product] = 1;
+            }
+        }
+
+        public Double GetPrice(Product product)
```
Comment (tavisca-mswarnesh) [inline AddToCart/Program.cs]: Double is also fine but i think it is unnecessary boxing

File: AddToCart/Program.cs [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Hello World!");
+        }
+    }
+
+    public class Product {
```
Comment (KumarWaghmode) [inline AddToCart/Program.cs]: This class should be in different file

File: AddToCart/Program.cs [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+        public double GetPrice(Product product)
+        {
+            try
+            {
+                return _cartItems[product] * product.Price - (_cartItems[product] * product.Price) * (double)product.Category / 100;
+            }
+            catch (Exception)
+            {
+                throw new EmptyCartException();
```
Comment (KumarWaghmode) [inline AddToCart/Program.cs]: Wrong exception

File: AddToCart/Program.cs [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+                return total - (total * _discount.GetDiscount(_discountCode)/100);
+            }
+            else
+            {
+                return GetTotal();
+            }
+        }
+    }
+
+    public class Discount {
```
Comment (KumarWaghmode) [inline AddToCart/Program.cs]: Move this class in different file

File: AddToCart/Program.cs [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+        public double GetDiscount(string discountCode)
+        {
+            if (discountCoupon.ContainsKey(discountCode))
+                return discountCoupon[discountCode];
+            else
+                return 0;
+        }
+    }
+
+    public enum Category
```
Comment (KumarWaghmode) [inline AddToCart/Program.cs]: Move this class in different file

File: AddToCart/Cart.cs [hunk 7]
```diff
@@ -0,0 +1,54 @@
+﻿namespace AddToCart
+{
+    public class Cart
+    {
+        public bool DiscountApplied { get; set; } = false;
```
Comment (KumarWaghmode) [inline AddToCart/Cart.cs]: Why is this public property??

File: AddToCart/CartItem.cs [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+        public double GetPrice(Product product,double discount=0)
+        {
+            try
+            {
+                return _cartItems[product] * product.Price - _cartItems[product] * product.Price*discount/100;
+            }
+            catch (Exception)
+            {
+                throw new EmptyCartException();
```
Comment (KumarWaghmode) [inline AddToCart/CartItem.cs]: This is wrong exception

File: AddToCart/CategoryDiscount.cs [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+            else
+                return 0;
+        }
+
+        public void AddDiscount(string category, double discount)
+        {
+            _category.Add(category,discount);
+        }
+
+        public void UpdateDiscount(string category, double updatedDiscount)
```
Comment (KumarWaghmode) [inline AddToCart/CategoryDiscount.cs:35]: What is use case of this method??

File: AddToCart/CategoryList.cs [hunk 10]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace AddToCart
+{
+    public enum CategoryList
+    {
+        grocery,
```
Comment (KumarWaghmode) [inline AddToCart/CategoryList.cs]: Enum value should be Pascal

**Review by KumarWaghmode — CHANGES_REQUESTED** [review]

**Pull request closed without merging**